cranelift/filetests/filetests/isa/s390x/call.clif

```diff
@@ -48,6 +48,7 @@ block0(v0: i32):
 ; block0:
 ;   lgr %r5, %r2
 ;   llgfr %r2, %r5
+;   lgr %r2, %r5
```

Comment:
Hi @elliottt this seems a bug: the `lgr` overwrites the effect of the `llgfr`.   Is there some problem with the extension handling on return values?

Reply:
@uweigand could you take another look? After the change to introduce a temporary for the extension, these tests also lack the initial `lgr` instruction, turning into just `llgfr %r2, %r2`.

Reply:
That's even better, removes an unnecessary copy and just zero-extends `%r2` in place.   The s390x test changes look all good to me now.  (Mostly just renamed registers; in several cases we get better code by eliminating register copies or spills, and a few cases -notably relating to divide instructions- we get slightly worse code.  Overall still good though.)